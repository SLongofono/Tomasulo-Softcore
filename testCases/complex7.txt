  # of rs Cycles in EX  Cycles in Mem # of FUs
Integer adder 2 1   1
FP adder  3 3   1
FP multiplier 2 20    1
Load/store unit 3 1 4 1
ROB entries = 128
CDB buffer entries = 1
R1=4,R2=20,R3=16,F2=30.1
Mem[0]=1.01,Mem[4]=2.01,Mem[8]=3.01,Mem[12]=4.01

ADD.D F2,F0,F0
ADD R2,R0,R0
LD F1,0(R2)
ADDI R2,R2,4
ADDI R1,R1,-1
ADD.D F2,F2,F1
BNE R0,R1,-5
SD F2,0(R3)
