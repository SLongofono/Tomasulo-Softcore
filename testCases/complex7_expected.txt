================================Instruction Completion Table====================
ID	| IS		 EX		 MEM		 WB		 COM
0	| 0		 1		 None		 4		 5
1	| 1		 2		 None		 3		 6
2	| 2		 3		 4		 8		 9
3	| 3		 4		 None		 5		 10
4	| 4		 5		 None		 6		 11
5	| 5		 9		 None		 12		 13
6	| 6		 7		 None		 None		 14
7	| 7		 8		 9		 13		 15
8	| 8		 9		 None		 10		 16
9	| 9		 10		 None		 11		 17
10	| 10		 14		 None		 17		 18
11	| 11		 12		 None		 None		 19
12	| 12		 13		 14		 18		 20
13	| 13		 14		 None		 15		 21
14	| 14		 15		 None		 16		 22
15	| 15		 19		 None		 22		 23
16	| 16		 17		 None		 None		 24
17	| 17		 18		 19		 22		 25
18	| 18		 19		 None		 20		 26
19	| 19		 20		 None		 21		 27
20	| 20		 24		 None		 27		 28
21	| 21		 22		 None		 None		 29
22      | 24             28              29              33              34

================================Integer ARF=====================================
R0 : 0              R1 : 0              R2 : 16             R3 : 16             
R4 : 0              R5 : 0              R6 : 0              R7 : 0              
R8 : 0              R9 : 0              R10: 0              R11: 0              
R12: 0              R13: 0              R14: 0              R15: 0              
R16: 0              R17: 0              R18: 0              R19: 0              
R20: 0              R21: 0              R22: 0              R23: 0              
R24: 0              R25: 0              R26: 0              R27: 0              
R28: 0              R29: 0              R30: 0              R31: 0              

================================Floating Point ARF==============================
F0 : 0.000000                           F1 : 4.010000                           
F2 : 10.40000                           F3 : 0.000000                           
F4 : 0.000000                           F5 : 0.000000                           
F6 : 0.000000                           F7 : 0.000000                           
F8 : 0.000000                           F9 : 0.000000                           
F10: 0.000000                           F11: 0.000000                           
F12: 0.000000                           F13: 0.000000                           
F14: 0.000000                           F15: 0.000000                           
F16: 0.000000                           F17: 0.000000                           
F18: 0.000000                           F19: 0.000000                           
F20: 0.000000                           F21: 0.000000                           
F22: 0.000000                           F23: 0.000000                           
F24: 0.000000                           F25: 0.000000                           
F26: 0.000000                           F27: 0.000000                           
F28: 0.000000                           F29: 0.000000                           
F30: 0.000000                           F31: 0.000000                           


================================Memory Unit=====================================
Word 00: 1.010000                       Word 01: 2.010000                       
Word 02: 3.010000                       Word 03: 4.010000                       
Word 04: 10.40000
